### DISCUSSION TITLE
Font requirements on Linux

### DISCUSSION THREAD

**Discussion by mj2015** [SixLabors/Fonts#200] [Q&A]
Could you explain the actual font requirements a little please? I have code that renders a font fine in Windows with Arial, with no loading or anything. On Linux, I apparently need to create a collection and install fonts. But the font files I have, which work fine in other applications, show an error `Invalid glyph format, only TTF glyph outlines supported`. The file is OTF, and I used a web converted to make a WOFF version, but same error. 

I'm hoping that someone can tell me what the various font formats within a font file are, and how I can tell which I have. And then if there is a way to convert them to a required format. Or, is there a way to get the font "from Linux" and use that? Unfortunately I have to use this specific font, so I am keen to learn what I need to do to make it work. Many thanks.

**Comment by mj2015** [comment]
Well, I've found the `SystemFonts` collection, and it shows some, but not all, of the fonts on the system. I guess this does confirm some format variation that I don't understand. Using a system font has my code working, so that's a step forward anyway.

**Comment by JimBobSquarePants** [comment]
Sounds like you might have been using the wrong method `AddCollection` to load the font that is designed for *.TTC files or maybe an OTF font that doesn't use True Type Glyphs.

I've been doing some cleanup in preparation for an RC to make the API more user friendly and augmented each method with a lot of documentation. The packages from MyGet are much closer to the API that will be in the v1 release.

https://github.com/SixLabors/Fonts/blob/998a0dad7a7fe6038b2eda42253bfeda282afd18/src/SixLabors.Fonts/IFontCollection.cs#L63-L76

System fonts on Linux are currently loaded from the following locations.
https://github.com/SixLabors/Fonts/blob/998a0dad7a7fe6038b2eda42253bfeda282afd18/src/SixLabors.Fonts/SystemFontCollection.cs#L31-L34

**Reply by mj2015** [reply]
"maybe an OTF font that doesn't use True Type Glyphs." - that's what I'm asking about. What is the font requirement, and how would I find out what sort of font I have?
